Rakefile

```diff
@@ -71,6 +75,63 @@ def re_run_basic_smoke_test
   sh beaker
 end
 
+def get_passing_json(current_branch_trunk)
+  jenkins_url = "https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/#{current_branch_trunk}/view/Suite/job/platform_puppet-agent_intn-van-promote_suite-daily-promotion-#{current_branch_trunk}/lastSuccessfulBuild/api/json"
+  @memoized_passing_json ||= Net::HTTP.get(URI.parse(jenkins_url))
```

Comment:
I don't see a place were we get the JSON and then act on it directly. It seems like everywhere we call this we then parse it with `JSON.parse`. In that case it might be valuable to parse it here and when robust-ifying we can then add the potential parsing error handling in one place.